frontend/openchat-push/src/push_sw.ts

```diff
@@ -127,136 +127,136 @@ async function showNotification(notification: Notification, id: string): Promise
     // trigger an alert. If false, and there is already at least one notification with a matching path, then this new
     // notification will be silent
     let closeExistingNotifications = false;
-    if (notification.kind === "direct_notification") {
+    if (n.kind === "direct_notification") {
         const chatId: DirectChatIdentifier = {
             kind: "direct_chat",
-            userId: notification.sender.userId,
+            userId: n.sender.userId,
         };
-        title = notification.senderName;
+        title = n.senderDisplayName ?? n.senderName;
```

Comment:
why do we send back two things (displayThing ?? thing) when all we just do is decide which one to display. Why can't the back end just send back whatever we are going to display so we don't have to do that?

Reply:
Well this is moot. We now pass in sender display_name in addition to sender username into various send message functions so that the notification message can be constructed. It so happens the client has decided to only render one thing but it might choose to render both display name and @username in the notifcations. On the other hand we should probably just remove the new display_name args and instead the client can pass (display_name ?? username) into the username field and the notifications can just use username as before.

Reply:
On second thoughts I think it is better to keep it the way it has been done. For direct chats we don't need to send the username/display name because the user canister already knows them and we don't want the user canister to decide how to render displayname+username.